### PR TITLE
feat: reset columns option

### PR DESCRIPTION
**Pull request by casperstorm**
## Proposed Changes
  - Adds a reset columns button in settings. 

## Checklist

- [ ] Tested on Windows
- [ ] Tested on MacOS
- [ ] Tested on Linux
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users

### CODE REVIEW

File: src/gui/update.rs [hunk 1]
```diff
@@ -230,6 +230,13 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
...
             return Ok(Command::perform(select_directory(), message));
         }
+        Message::Interaction(Interaction::ResetColumns) => {
+            log::debug!("Interaction::ResetColumns");
+
+            ajour.config.column_config = Default::default();
+            apply_config(ajour, ajour.config.clone());
+            let _ = &ajour.config.save();
+        }
         Message::Interaction(Interaction::OpenLink(link)) => {
```
Comment (casperstorm) [inline src/gui/update.rs:240]: @tarkah, currently when pressing the button and executing this code and closing and reopening Ajour the columns is reset, however I am wondering why it does not happen instantly and appear changed in GUI?

File: src/gui/mod.rs [hunk 2]
```diff
@@ -1987,7 +1991,7 @@ async fn load_caches() -> Result<(FingerprintCache, AddonCache)> {
     Ok((fingerprint_cache, addon_cache))
 }
 
-fn apply_config(ajour: &mut Ajour, config: Config) {
+pub fn apply_config(ajour: &mut Ajour, config: Config) {
```
Comment (casperstorm) [inline src/gui/mod.rs:1994]: How do you like this, @tarkah? I would need to access `apply_config` from `update`.

Comment (tarkah) [reply src/gui/mod.rs:1994]: Yeah I think that makes sense! No reason to reinvent the wheel

File: src/gui/update.rs (hunk 1 above)
Comment (tarkah) [reply src/gui/update.rs:240]: Let me test and see if I can figure it out

**Comment by tarkah** [comment]
@casperstorm see my commit, I've fixed it. The problem is when you set the config.column_settings to Default, it just creates an empty HashMap. So when you did the `apply_config` function, it didn't reset the internal states correctly.

The proper order is to reset all the internal column state structs to Default then serialize those Defaults back to the config file.

**Comment by casperstorm** [comment]
> @casperstorm see my commit, I've fixed it. The problem is when you set the config.column_settings to Default, it just creates an empty HashMap. So when you did the `apply_config` function, it didn't reset the internal states correctly.
> 
> 
> 
> The proper order is to reset all the internal column state structs to Default then serialize those Defaults back to the config file.

Ah of cause. Let's get this into today's RC! Awesome you nailed it.

**Comment by tarkah** [comment]
Sweet! This is good to merge then when you're ready

**Pull request merged**